examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -470,14 +465,18 @@ class _DiamondFabState extends State<_DiamondFab> {
     //  the host's top edge where the notch starts (marked with "*").
     //  We compute notchToCenter by similar triangles:
     final double notchToCenter =
-      intersection.height * (marginedGuest.height / 2.0)
-      / (marginedGuest.width / 2.0);
+      intersection.height * (guest.height / 2.0)
```

Comment:
Even if it's impossible for guest.height to be 0 here, might be best to assert as much before dividing by guest.height.

Reply:
Added an assertion that guest.width is greater than 0.